### PR TITLE
Related to #2114: reformat docstrings in the 'metrics' folder

### LINKED ISSUE #2114: Improve doc examples rendering
Currently we are rendering our examples in the docs like this https://pytorch.org/ignite/generated/ignite.engine.engine.Engine.html#ignite.engine.engine.Engine which is not very neat.
What we want to do is to find a better way to make it more good looking, and update our docs examples

image

- Replace "Examples" with some sphinx tag or something else, for example, like scikit-image does:

image

### PR DESCRIPTION
**Pull request by toxa23**
Related to #2114

Description:
part 6 of 7, reformat docstrings in the 'metrics' folder, make sure all examples have "Examples:" header, move "Args:" block to the top of method documentation

Check list:

- [ ] New tests are added (if a new feature is added)
- [X] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by KickItLikeShika** [comment]
There is an error with code formatting, we use flake8, black, and isort, please check [the formatting part in out contribution guide](https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#formatting-code).
The error is related to flake8 and the length of the lines

**Review by KickItLikeShika — COMMENTED** [review]
Thanks for the PR, Anton! I left a comment.

File: ignite/metrics/accuracy.py [hunk 1]
```diff
@@ -105,18 +105,7 @@ class Accuracy(_BaseClassification):
     - `y` and `y_pred` must be in the following shape of (batch_size, num_categories, ...) and
       num_categories must be greater than 1 for multilabel cases.
 
-    In binary and multilabel cases, the elements of `y` and `y_pred` should have 0 or 1 values. Thresholding of
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply ignite/metrics/accuracy.py:108]: Thanks @KickItLikeShika , yes, I agree with your comment. @toxa23 let's not split this part

**Review by KickItLikeShika — APPROVED** [review]
LGTM! Thanks, Anton!

**Pull request merged**